test/boost/cql_query_test.cc

```diff
@@ -153,6 +153,74 @@ SEASTAR_TEST_CASE(test_twcs_max_window) {
     });
 }
 
+SEASTAR_TEST_CASE(test_twcs_restrictions_mixed) {
+    return do_with_cql_env_thread([](cql_test_env& e) {
+        // Cenario 1: STCS->TWCS with no TTL defined
+        e.execute_cql("CREATE TABLE tbl (a int PRIMARY KEY, b int);").get();
+        e.execute_cql("ALTER TABLE tbl WITH compaction = {'class': 'TimeWindowCompactionStrategy'};").get();
+        e.require_table_exists("ks", "tbl").get();
+
+        // Cenario 2: STCS->TWCS with small TTL
+        e.execute_cql("CREATE TABLE tbl2 (a int PRIMARY KEY, b int) WITH default_time_to_live = 60;").get();
+        e.execute_cql("ALTER TABLE tbl2 WITH compaction = {'class': 'TimeWindowCompactionStrategy'};").get();
+        e.require_table_exists("ks", "tbl2").get();
+
+        // Cenario 3: STCS->TWCS with large TTL value
+        e.execute_cql("CREATE TABLE tbl3 (a int PRIMARY KEY, b int) WITH default_time_to_live = 8640000;").get();
+        e.require_table_exists("ks", "tbl3").get();
+        assert_that_failed(e.execute_cql("ALTER TABLE tbl3 WITH compaction = {'class': 'TimeWindowCompactionStrategy'};"));
+
+        // Cenario 4: TWCS table with small to large TTL
+        e.execute_cql("CREATE TABLE tbl4 (a int PRIMARY KEY, b int) WITH compaction = "
+                      "{'class': 'TimeWindowCompactionStrategy'} AND default_time_to_live = 60;").get();
+        e.require_table_exists("ks", "tbl4").get();
+        assert_that_failed(e.execute_cql("ALTER TABLE tbl4 WITH default_time_to_live = 86400000;"));
+
+        // Cenario 5: No TTL TWCS to large TTL and then small TTL
+        e.execute_cql("CREATE TABLE tbl5 (a int PRIMARY KEY, b int) WITH compaction = "
+                      "{'class': 'TimeWindowCompactionStrategy'}").get();
+        e.require_table_exists("ks", "tbl5").get();
+        assert_that_failed(e.execute_cql("ALTER TABLE tbl5 WITH default_time_to_live = 86400000;"));
+        e.execute_cql("ALTER TABLE tbl5 WITH default_time_to_live = 60;").get();
+
+        // Cenario 6: twcs_max_window_count LiveUpdate - Decrease TTL
+        e.execute_cql("UPDATE system.config SET value='0' WHERE name='twcs_max_window_count';").get();
+        e.execute_cql("CREATE TABLE tbl6 (a int PRIMARY KEY, b int) WITH compaction = "
+                      "{'class': 'TimeWindowCompactionStrategy'} AND default_time_to_live = 86400000;").get();
+        e.require_table_exists("ks", "tbl6").get();
+        e.execute_cql("UPDATE system.config SET value='50' WHERE name='twcs_max_window_count';").get();
```

Comment:
Isn't 50 the default of this parameter?

Reply:
Correct, I've added a smaller value at the beginning of the test so I didn't have to rewrite it all. Also - it prevents some sections from failing if we decided to change the default in the future.